### PR TITLE
html()/text() bugfixes

### PR DESCRIPTION
**Pull request by rvagg**
2 small fixes, will comment inline:

### CODE REVIEW

File: bonzo.js [hunk 1]
```diff
@@ -246,7 +246,7 @@
         }
         return typeof h !== 'undefined' ?
             this.each(function (el) {
-              (m = el.tagName.match(specialTags)) ?
+              !text && (m = el.tagName.match(specialTags)) ?
```
Comment (rvagg) [inline bonzo.js:249]: bypass `append()` for text updates or we'll end up going through `create()`

File: bonzo.js [hunk 2]
```diff
@@ -585,12 +585,13 @@
...
           , pn = parentNode
           , tb = features.autoTbody && p && p[0] == '<table>' && !(/<tbody/i).test(node)
+
         el.innerHTML = p ? (p[0] + node + p[1]) : node
         while (dep--) el = el.firstChild
         do {
           // tbody special case for IE<8, creates tbody on any empty table
           // we don't want it if we're just after a <thead>, <caption>, etc.
-          if (el.nodeType == 1 && (!tb || el.tagName.toLowerCase() != 'tbody')) {
+          if ((!tag || el.nodeType == 1) && (!tb || el.tagName.toLowerCase() != 'tbody')) {
```
Comment (rvagg) [inline bonzo.js:594]: if you're doing a `create()` without a &lt;tag&gt; then you don't get the result because `el.nodeType !== 1`.
An earlier version of Bonzo did it properly but this commit, changing a single line, would have stopped it working: https://github.com/ded/bonzo/commit/a4cf1b273f07a8e093ddaf9e56492780d26fd22a
I'm not sure exactly what that was for cause there aren't notes or extra tests, I'm guessing it was to help process tags surrounded by whitespace. Can you remember if there is a special case that commit was addressing that should be added to the tests?

**Comment by rvagg** [comment]
Just added some extra testing after that last comment of mine. Confirming that `create('   <p>surrounded by whitespace</p>    ')` works.

**Comment by ded** [comment]
Solid
